### ISSUE TITLE
Unable to use Databricks CLI Profiles

### ISSUE THREAD

**Issue by AxREki** [paiqo/Databricks-VSCode#108]
Hello,

I am struggling to configure the extension with Databricks CLI Profiles. Here is my .databrickscfg file I am currently using with the CLI (version 0.17.3) installed.

![image](https://user-images.githubusercontent.com/50666376/201783604-c4c84ad7-d301-4484-934a-705d89727d55.png)

When switch Databricks extension connection manager in VSCode to "Databricks CLI Profiles", the profiles do appear in VSCode but I can't see anything else.
![image](https://user-images.githubusercontent.com/50666376/201783853-4d2967be-9913-425f-b37e-ef9d7d1bebf7.png)
Even after refresh :
![image](https://user-images.githubusercontent.com/50666376/201783949-b5ce4295-3ab2-49a2-aa45-4f22538ecf2f.png)

When refreshing the profiles, I see the error : 
![image](https://user-images.githubusercontent.com/50666376/201784003-3ff14701-6d79-4801-85f3-a034d8d0da59.png)
But am unable to find where and how to fix it.

When hovering, I don't see the organization id on the URL. Could it be a lead ? 
![image](https://user-images.githubusercontent.com/50666376/201784348-702c2a19-d034-4508-a70e-b830c3ba8f05.png)

Thanks for your help.

**Comment by gbrueckl** [comment]
Can you try your second connectiton please (test-axr). It could be that it does not work well with the o=123xxx suffix at the end of the url.
I also recommend to use the very same url as in the browser which should also contain the organization id as part if the url.
Restarting vscode after those changes might also help

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: AxR ***@***.***>
Sent: Monday, November 14, 2022 11:47:33 PM
To: paiqo/Databricks-VSCode ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [paiqo/Databricks-VSCode] Unable to use Databricks CLI Profiles (Issue #108)


Hello,

I am struggling to configure the extension with Databricks CLI Profiles. Here is my .databrickscfg file I am currently using with the CLI installed.

[image]<https://user-images.githubusercontent.com/50666376/201783604-c4c84ad7-d301-4484-934a-705d89727d55.png>

When switch Databricks extension connection manager in VSCode to "Databricks CLI Profiles", the profiles do appear in VSCode but I can't see anything else.
[image]<https://user-images.githubusercontent.com/50666376/201783853-4d2967be-9913-425f-b37e-ef9d7d1bebf7.png>
Even after refresh :
[image]<https://user-images.githubusercontent.com/50666376/201783949-b5ce4295-3ab2-49a2-aa45-4f22538ecf2f.png>

When refreshing the profiles, I see the error :
[image]<https://user-images.githubusercontent.com/50666376/201784003-3ff14701-6d79-4801-85f3-a034d8d0da59.png>
But am unable to find where and how to fix it.

Thanks for your help.

**Comment by AxREki** [comment]
Thank for the tip. It lead me to find an explanation.
The second connection "axr-test" didn't work as-is but works if I find the "real" url, not the one in my browser.
![image](https://user-images.githubusercontent.com/50666376/201929449-20ede7f4-a7db-4f42-b2b5-e71f41712872.png)
On the Azure Databricks, my url is always https://<region_name>.azuredatabricks.net/?o=<org_id>
But when I get the url from the spark context with ```spark.conf.get('spark.databricks.workspaceUrl')```, it gives me the one with the format ```https://adb-<org_id>.XX.azuredatabricks.net``` (XX being a random number...!)

Is there a chance we could get it working with the o=123 format ? I have around 60 workspace right now and can't just find the url because of the random number in it. No emergency of course. And maybe I can try to submit a PR. What would you recommend ?

Thanks again !
To recap :
![image](https://user-images.githubusercontent.com/50666376/201931241-ecf47a1f-a050-44bd-aaeb-84d216428419.png)

**Comment by gbrueckl** [comment]
I would definitely recommend to use the full URL that is provided by the Azure portal which is `https://adb-<org_id>.XX.azuredatabricks.net` 

I will also add a check to remove everything in the URL thats behind the authority automatically 

regarding Azure, we are planning to leverage the Azure Account authentication to get a list of available workspaces and show them automatically in the connection dialog, but this will only be done in 2023 as it seems at the moment

**Comment by gbrueckl** [comment]
just released `v1.3.0` which now also offers a new Azure connection manager!
https://github.com/paiqo/Databricks-VSCode/tree/v1.3.0#setup-and-configuration-azure-connection-manager

**Comment by gbrueckl** [comment]
closing as fixed
